cts-java/src/test/java/org/ga4gh/cts/api/reads/ReadsSearchIT.java

```diff
@@ -0,0 +1,263 @@
+            final SearchReadGroupSetsResponse resp = client.reads.searchReadGroupSets(req);
+
+            final List<ReadGroupSet> readGroupSets = resp.getReadGroupSets();
+
+            assertThat(readGroupSets).hasSize(1);
+            final ReadGroupSet readGroupSet = readGroupSets.get(0);
+            assertThat(readGroupSet.getName()).isEqualTo(expectedReadGroupSetName);
+            assertThat(readGroupSet.getDatasetId()).isEqualTo(TestData.getDatasetId());
+        }
+    }
+
+    /**
+     * Verify that passing zero read group names in a {@link SearchReadsRequest} fails.
+     *
+     * @throws Exception if there's a problem we didn't catch
+     */
+    @SuppressWarnings("ThrowableResultOfMethodCallIgnored")
+    @Test
+    public void searchReadsWithNoIdsFails() throws Exception {
+
+        // first get a valid reference
+        final String refId = Utils.getValidReferenceId(client);
+
+        final SearchReadsRequest request =
+                SearchReadsRequest.newBuilder()
+                                  .setReadGroupIds(Collections.emptyList())
+                                  .setStart(0L)
+                                  .setEnd(150L)
+                                  .setReferenceId(refId)
+                                  .build();
+
+        final GAWrapperException t = catchGAWrapperException(() -> client.reads.searchReads(request));
+        assertThat(t.getHttpStatusCode()).isEqualTo(HttpURLConnection.HTTP_NOT_IMPLEMENTED);
+    }
+
+    /**
+     * Verify that passing one read group name in a {@link SearchReadsRequest}
+     * returns valid reads.
+     *
+     * @throws Exception if there's a problem
+     */
+    @Test
+    public void searchReadsWithOneReadGroupIdSucceeds() throws Exception {
+
+        // first get a valid reference
+        final String refId = Utils.getValidReferenceId(client);
+
+        // get a ReadGroup id
+        final String readGroupId = Utils.getReadGroupId(client);
+
+        final SearchReadsRequest request =
+                SearchReadsRequest.newBuilder()
+                                  .setReadGroupIds(aSingle(readGroupId))
+                                  .setStart(0L)
+                                  .setEnd(150L)
+                                  .setReferenceId(refId)
+                                  .build();
+        // XXX fails due to server issue #591
+        final SearchReadsResponse response = client.reads.searchReads(request);
+
+        assertThat(response.getAlignments()).isNotNull();
+
+        response.getAlignments().stream()
+                .forEach(readAlignment ->
+                                 assertThat(readAlignment.getAlignedSequence()).isNotNull()
+                                                                               .matches("[ACTGN]+"));
+    }
+
+    /**
+     * Verify that passing all known read group names in a {@link SearchReadsRequest}
+     * returns all matching read groups.
+     * XXX fails due to server issue #596
```

Comment:
Does this belong here? I generally read 'XXX' as 'resolve this issue before submitting'.